src/detectors/ZDC/ZDC.cc

```diff
@@ -102,5 +106,150 @@ extern "C" {
             app   // TODO: Remove me once fixed
           )
         );
+
+        //side length of hexagonal cells in SiPM-on-tile part of the ZDC
+        auto side_length = 31.3 * dd4hep::mm;
+
+        app->Add(new JOmniFactoryGeneratorT<CalorimeterHitDigi_factory>(
+          "ZDCRawHits", {"HcalFarForwardZDCHits"}, {"ZDCRawHits"},
+          {
+            .tRes = 0.0 * dd4hep::ns,
+            .capADC = 32768,
+            .dyRangeADC = 800 * dd4hep::MeV,
+            .pedMeanADC = 400,
+            .pedSigmaADC = 10,
+            .resolutionTDC = 10 * dd4hep::picosecond,
+            .corrMeanScale = 1.0,
+          },
+          app   // TODO: Remove me once fixed
+        ));
+
+        app->Add(new JOmniFactoryGeneratorT<CalorimeterHitReco_factory>(
+          "ZDCRecHits", {"ZDCRawHits"}, {"ZDCRecHits"},
+          {
+            .capADC = 32678,
+            .dyRangeADC = 800. * dd4hep::MeV,
+            .pedMeanADC = 400,
+            .pedSigmaADC = 10,
+            .resolutionTDC = 10 * dd4hep::picosecond,
+            .thresholdFactor = 0.0,
+            .thresholdValue = -100.0,
+            .sampFrac = 1.0,
+            .readout = "HcalFarForwardZDCHits",
+            .layerField = "layer",
+            .sectorField = "system",
+          },
+          app   // TODO: Remove me once fixed
+        ));
+
+        app->Add(new JOmniFactoryGeneratorT<HEXPLIT_factory>(
+          "ZDCSubcellHits", {"ZDCRecHits"}, {"ZDCSubcellHits"},
+          {
+            .layer_spacing=25.1*dd4hep::mm,
```

Comment:
Geometry information should come from geometry, not a parameter in reconstruction.

Reply:
@wdconinc can you point me to somewhere in the code where I can obtain a named parameter?   I assume that I can have it parse through the detector description xml to get that information; is that correct?

Reply:
Yes, there are some examples at:
https://github.com/eic/EICrecon/blob/0a27c9a09b52f02e9b6888e9d75003d45243ce6c/src/services/geometry/richgeo/ReadoutGeo.cc#L50-L55

Reply:
thanks.  I have gotten this to work without ever using layer_spacing (by using layer number comparisons) nor side_length (which I have instead used the cell dimensions of the hit to extract).